sources/packages/backend/apps/api/src/route-controllers/audit/audit.controller.ts

```diff
@@ -25,14 +26,20 @@ export class AuditController extends BaseController {
    * @param userToken user token.
    * @param request request object.
    */
-  @Post(":event")
+  @Post()
   audit(
-    @Param("event", new ParseEnumPipe(AuditEvent)) event: AuditEvent,
+    @Body() payload: AuditAPIInDTO,
     @UserToken() userToken: IUserToken,
     @Req() request: Request,
   ): void {
     const clientIp =
-      request.headers["x-forwarded-for"] || request.socket.remoteAddress;
-    this.auditService.audit(clientIp, userToken.userName, event, userToken.azp);
+      (request.headers["x-forwarded-for"] as string) ||
```

Comment:
For my understanding, why did we need this? Was something fixed as a result of this?

Reply:
If the connection is direct between client and server `request.socket.remoteAddress` is used but if you're running the application behind a proxy, you might need to get the IP address from `X-Forwarded-For` headers.
The conversion of `request.headers["x-forwarded-for"]` to string is because [IncomingHttpHeaders](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/v18/http.d.ts#L49) extends NodeJS.Dict<string | string[]> but in practice it's a string with comma separated IPs.